File: leetcode/DynamicProgram/leetcode-70/70.go
package dynamicprogram

func climbStairs(n int) int {
	if n < 2 {
		return n
	}
	p, q := 1, 1
	for i := 2; i <= n; i++ {
		p, q = q, p+q
	}
	return q
}

File: leetcode/DynamicProgram/leetcode-70/70.java
class Solution {
    public int climbStairs(int n) {
        if (n < 2) {
            return n;
        }
        int p = 1, q = 1;
        for (int i = 2; i <= n; i++) {
            int tmp = q;
            q = p + q;
            p = tmp;
        }
        return q;
    }
}